### PR TITLE
Fix Button broken display for loading and checked status with children

### PR DESCRIPTION
**Pull request by nickoola**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Unreleased
 
 - **[UPDATE]** Added aria attributes and focus trap on Modals
+- **[FIX]** `Button` do not display children on loading and checked status
```
Comment (mcellier) [inline CHANGELOG.md]: `Button` doesn't display children on `LOADING` and `CHECKED` statuses

File: src/button/index.tsx [hunk 2]
```diff
@@ -189,7 +189,7 @@ export default class Button extends PureComponent<ButtonProps, ButtonState> {
             onDoneAnimationEnd={onDoneAnimationEnd}
           />
         )}
-        {children}
+        {!hasLoader && children}
```
Comment (mcellier) [inline src/button/index.tsx]: hasLoader does the job for LOADING & CHECKED ?

Comment (mcellier) [reply src/button/index.tsx]: Add unit tests

Comment (nickoola) [reply src/button/index.tsx]: Yep Done

File: CHANGELOG.md (hunk 1 above)
Comment (nickoola) [reply CHANGELOG.md]: Done

Comment (Tuxosaurus) [reply CHANGELOG.md]: Sorry I'm late, I would say "shouldn't display", else I'm not sure if we're talking about the fix or the problem

File: src/button/index.tsx (hunk 2 above)
Comment (nickoola) [reply src/button/index.tsx]: And should have a discussion about the collapsed display. I agree hasLoader is a bit more than just managing loading state.

**Pull request merged**